library/Icingadb/Util/PerfData.php

```diff
@@ -0,0 +1,546 @@
+        }
+        if ($this->criticalThreshold === null) {
+            $this->criticalThreshold = new ThresholdRange();
+        }
+    }
+
+    /**
+     * Return the given value converted to its smallest supported representation
+     *
+     * @param   string      $value      The value to convert
+     * @param   string      $fromUnit   The unit the value currently represents
+     *
+     * @return  null|float              Null in case the value is not a number
+     */
+    protected static function convert($value, $fromUnit = null)
+    {
+        if ($value instanceof ThresholdRange) {
+            $value = clone $value;
+
+            $min = $value->getMin();
+            if ($min !== null) {
+                $value->setMin(self::convert($min, $fromUnit));
+            }
+
+            $max = $value->getMax();
+            if ($max !== null) {
+                $value->setMax(self::convert($max, $fromUnit));
+            }
+
+            return $value;
+        }
+
+        if (is_numeric($value)) {
+            switch ($fromUnit) {
+                case 'us':
+                    return $value / pow(10, 6);
+                case 'ms':
+                    return $value / pow(10, 3);
+                case 'tb':
+                    return floatval($value) * pow(2, 40);
+                case 'gb':
+                    return floatval($value) * pow(2, 30);
+                case 'mb':
+                    return floatval($value) * pow(2, 20);
+                case 'kb':
+                    return floatval($value) * pow(2, 10);
+                default:
+                    return (float) $value;
+            }
+        }
+    }
+
+    protected function calculatePieChartData()
+    {
+        $rawValue = $this->getValue();
+        $minValue = $this->getMinimumValue() !== null ? $this->getMinimumValue() : 0;
+        $usedValue = ($rawValue - $minValue);
+
+        $green = $orange = $red = 0;
+
+        if ($this->criticalThreshold->contains($rawValue)) {
+            if ($this->warningThreshold->contains($rawValue)) {
+                $green = $usedValue;
+            } else {
+                $orange = $usedValue;
+            }
+        } else {
+            $red = $usedValue;
+        }
+
+        return array($green, $orange, $red, ($this->getMaximumValue() - $minValue) - $usedValue);
+    }
+
+
+    public function asInlinePie()
+    {
+        if (! $this->isVisualizable()) {
+            throw new ProgrammingError('Cannot calculate piechart data for unvisualizable perfdata entry.');
+        }
+
+        $data = $this->calculatePieChartData();
+        $pieChart = new InlinePie($data, $this);
+        $pieChart->setColors(array('#44bb77', '#ffaa44', '#ff5566', '#ddccdd'));
+
+        return $pieChart;
+    }
+
+    /**
+     * Format the given value depending on the currently used unit
+     */
+    protected function format($value)
+    {
+        if ($value instanceof ThresholdRange) {
```

Comment:
Please add the same check as in https://github.com/Icinga/icingaweb2/pull/4351. Empty values should show `-`, not `0.00`.